plugin/pkg/admission/podtolerationrestriction/admission_test.go

```diff
@@ -230,57 +215,74 @@ func TestPodAdmission(t *testing.T) {
 		},
 	}
 	for _, test := range tests {
-		if test.namespaceTolerations != nil {
-			tolerationStr, err := json.Marshal(test.namespaceTolerations)
-			if err != nil {
-				t.Errorf("error in marshalling namespace tolerations %v", test.namespaceTolerations)
+		t.Run(test.testName, func(t *testing.T) {
+			namespace := &api.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        "testNamespace",
+					Namespace:   "",
+					Annotations: map[string]string{},
+				},
 			}
-			namespace.Annotations = map[string]string{NSDefaultTolerations: string(tolerationStr)}
-		}
 
-		if test.whitelist != nil {
-			tolerationStr, err := json.Marshal(test.whitelist)
-			if err != nil {
-				t.Errorf("error in marshalling namespace whitelist %v", test.whitelist)
+			if test.namespaceTolerations != nil {
+				tolerationStr, err := json.Marshal(test.namespaceTolerations)
+				if err != nil {
+					t.Errorf("error in marshalling namespace tolerations %v", test.namespaceTolerations)
+				}
+				namespace.Annotations = map[string]string{NSDefaultTolerations: string(tolerationStr)}
 			}
-			namespace.Annotations[NSWLTolerations] = string(tolerationStr)
-		}
 
-		informerFactory.Core().InternalVersion().Namespaces().Informer().GetStore().Update(namespace)
+			if test.whitelist != nil {
+				tolerationStr, err := json.Marshal(test.whitelist)
+				if err != nil {
+					t.Errorf("error in marshalling namespace whitelist %v", test.whitelist)
+				}
+				namespace.Annotations[NSWLTolerations] = string(tolerationStr)
+			}
 
-		handler.pluginConfig = &pluginapi.Configuration{Default: test.defaultClusterTolerations, Whitelist: test.clusterWhitelist}
-		pod := test.pod
-		pod.Spec.Tolerations = test.podTolerations
+			mockClient := fake.NewSimpleClientset(namespace)
```

Comment:
For those attempting to review. This line is the only difference between the old test and the new test other than moving it into a closure. The following line

```
mockClient := fake.NewSimpleClientset(namespace)
```

Replaces

```
informerFactory.Core().InternalVersion().Namespaces().Informer().GetStore().Update(namespace)
```

We found that the informer would occasionally not return the updated namespace. When the admission controller can't find the object in the informer, it calls the fake clientset which would return an empty namespace object.